pkg/controller/pod_test.go

```diff
@@ -221,6 +224,48 @@ func TestPodReconcile(t *testing.T) {
 				require.NoError(t, err)
 			},
 		},
+		{
+			description: "[Success] Should remove entry from podToClaimsMap and claimToPodMap if pod has been deleted.",
+			request:     testPod0Request,
+			setupFunc: func(t *testing.T, mockCtl *gomock.Controller) *ReconcilePod {
+				newAttachment := testPrimaryAzVolumeAttachment0.DeepCopy()
+				newAttachment.Status.State = azdiskv1beta2.Attached
+
+				newVolume := testAzVolume0.DeepCopy()
+				newVolume.Status.Detail = &azdiskv1beta2.AzVolumeStatusDetail{
+					VolumeID: testManagedDiskURI0,
+				}
+
+				controller := NewTestPodController(
+					mockCtl,
+					testNamespace,
+					newVolume,
+					newAttachment,
+					&testNode0,
+					&testNode1,
+					&testPersistentVolume0)
```

Comment:
It would be good to actually register a pod object, run `addPod`, and delete the registered pod to make sure that the next reconcile removes entry added from `addPod`.